File: Leetcode/Daily/06-2025/Java/21.java
// Problem: 3085. Minimum Deletions to Make String K-Special (Medium)
// Link: https://leetcode.com/problems/minimum-deletions-to-make-string-k-special

class Solution {
    public int minimumDeletions(String word, int k) {
        int[] freq = new int[26];
        int n = word.length();

        int ans = Integer.MAX_VALUE;
        for (int i = 0; i < n; i++) freq[word.charAt(i) - 'a']++;
        
        Arrays.sort(freq);

        for (int i = 25; i >= 0; i--) {
            int temp = 0;
            for (int j = i-1; j >= 0; j--) {
                temp += freq[j];
            }
            int target = freq[i] + k;
            for (int j = i; j < 26; j++) {
                if (freq[j] <= target) continue;
                temp += freq[j] - target;
            }
            ans = Math.min(ans, temp);
        }

        return ans;        
    }
}

File: Leetcode/Daily/06-2025/Python3/21.py
// Problem: 3085. Minimum Deletions to Make String K-Special (Medium)
// Link: https://leetcode.com/problems/minimum-deletions-to-make-string-k-special

class Solution:
    def minimumDeletions(self, word: str, k: int) -> int:
        freq = [0] * 26
        for ch in word:
            freq[ord(ch) - ord('a')] += 1

        freq.sort()
        ans = 2**31 -1
        
        for i in range(25, -1, -1):
            temp = 0
            for j in range(i-1, -1, -1):
                temp += freq[j]

            target = freq[i] + k
            for j in range(i, 26):
                if freq[j] <= target:
                    continue
                temp += freq[j] - target
            
            ans = min(ans, temp)
        

        return ans;        